packages/react/fast-components-react-base/src/data-grid/data-grid.tsx

```diff
@@ -0,0 +1,1103 @@
+        ) {
+            shouldUpdateState = true;
+            focusColumnKey = this.props.defaultFocusColumnKey;
+        }
+
+        // apply a new focus rowkey if one is provided and it exists
+        if (
+            this.props.defaultFocusRowKey !== prevProps.defaultFocusRowKey &&
+            !isNil(this.props.defaultFocusRowKey)
+        ) {
+            const newFocusRowIndex: number = this.getRowIndexByKey(
+                this.props.defaultFocusRowKey
+            );
+
+            if (newFocusRowIndex !== -1) {
+                shouldUpdateState = true;
+                focusRowKey = this.props.defaultFocusRowKey;
+                desiredVisibleRowIndex = newFocusRowIndex;
+                focusRowIndex = newFocusRowIndex;
+                currentDataPageStartIndex =
+                    newFocusRowIndex - Math.floor(this.props.pageSize / 2);
+                if (currentDataPageStartIndex < 0) {
+                    currentDataPageStartIndex = 0;
+                }
+                currentDataPageEndIndex = Math.min(
+                    currentDataPageStartIndex + this.props.pageSize,
+                    this.props.gridData.length - 1
+                );
+            }
+        }
+
+        // revalidate when there is new data
+        if (this.props.gridData !== prevProps.gridData) {
+            shouldUpdateState = true;
+
+            if (this.props.gridData.length === 0) {
+                rowPositions = [];
+                currentDataPageStartIndex = -1;
+                currentDataPageEndIndex = -1;
+                estimatedTotalHeight = 0;
+                focusRowKey = null;
+                focusColumnKey = null;
+                focusRowIndex = -1;
+            } else {
+                // if an author tells us nothing has changed before a certain point we keep position data
+                // up to that point.
+                rowPositions = rowPositions.slice(0, this.props.stableRangeEndIndex);
+
+                // ensure focus is still valid
+                const newFocusRowIndex: number = this.getRowIndexByKey(focusRowKey);
+                if (newFocusRowIndex === -1) {
+                    // our focus row no longer exists, assign a new one based on previous focus row index
+                    focusRowIndex = Math.min(
+                        Math.max(focusRowIndex, 0),
+                        this.props.gridData.length - 1
+                    );
+                    focusRowKey = this.props.gridData[focusRowIndex][
+                        this.props.dataRowKey
+                    ];
+                } else {
+                    focusRowIndex = newFocusRowIndex;
+                }
+
+                // ensure data page conforms to new data length
+                currentDataPageStartIndex =
+                    focusRowIndex - Math.floor(this.props.pageSize / 2);
+
+                currentDataPageStartIndex =
+                    currentDataPageStartIndex < 0 ? 0 : currentDataPageStartIndex;
```

Comment:
seems like you can do something like `currentDataPageStartIndex = Math.max(0, focusRowIndex - Math.floor(this.props.pageSize/2);` instead of repeatedly checking to see if the number is less than 0.